llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -1949,24 +1955,48 @@ bool SPIRVInstructionSelector::selectSign(Register ResVReg,
   return Result;
 }
 
+bool SPIRVInstructionSelector::selectWaveNOpInst(Register ResVReg,
+                                                 const SPIRVType *ResType,
+                                                 MachineInstr &I,
+                                                 unsigned Opcode,
+                                                 unsigned OperandCount) const {
+  assert(I.getNumOperands() == OperandCount);
```

Comment:
Why are you even taking in `OperandCount` as a value just use  `getNumOperands`.

Reply:
I was just blindly refactoring, but one could argue that asserting the expected `OperandCount` is safer from a "is this doing what I expect" angle

Reply:
Number of operands per opcode is already determined in the tablegen. Nothing is going to change that. This just adds a pain point for users of this function where they think they can specify the operand length first build and test fails and then they do what tablegen defined on the second build.  We should avoid creating apis that can mislead users.

Reply:
https://github.com/llvm/llvm-project/pull/115902/commits/42d24e47bfd0d635119bfa858bdf01b14238fe8c